engine/target_sym_engine.ml

```diff
@@ -11,7 +11,7 @@ module IMap = Map.Make(struct type t = int let compare = compare end)
 type constraint_tree =
   | Failure
   | Leaf of target_blackbox
-  | Node of (pi list * constraint_tree) list
+  | Node of (pi * constraint_tree) list * (pi list * constraint_tree) option
```

Comment:
Note: in this case `None` means `Unreachable`, not `Failure`.

Reply:
yes, but we don't have an Unreachable type. Should we add it?